Dockerfile.test.cpu

```diff
@@ -98,8 +98,10 @@ RUN pip install --no-cache-dir ray==1.7.0
 
 # Install MPI.
 RUN if [[ ${MPI_KIND} == "OpenMPI" ]]; then \
-        wget --progress=dot:mega -O /tmp/openmpi-3.0.0-bin.tar.gz https://github.com/horovod/horovod/files/1596799/openmpi-3.0.0-bin.tar.gz && \
-            cd /usr/local && tar -zxf /tmp/openmpi-3.0.0-bin.tar.gz && ldconfig && \
+        wget --progress=dot:mega -O /tmp/openmpi-4.1.4-bin.tar.gz https://download.open-mpi.org/release/open-mpi/v4.1/openmpi-4.1.4.tar.gz && \
+            cd /tmp && tar -zxf /tmp/openmpi-4.1.4-bin.tar.gz && \
+            mkdir openmpi-4.1.4/build && cd openmpi-4.1.4/build && ../configure --prefix=/usr/local && \
+            make -j all && make install && ldconfig && \
```

Comment:
@romerojosh I was using the same set of commands to build the image with OpenMPI 4.1.4 on my mac. However, when I start a container with that image and run tests I am facing the illegal instruction issue as mentioned in https://github.com/horovod/horovod/pull/3662#issuecomment-1228885412 . The error is surprising as we are not facing it in the CI runs.

Reply:
```ipython
In [1]: import torch
Extension horovod.torch has not been built: /horovod/horovod/torch/mpi_lib_v2.cpython-38-x86_64-linux-gnu.so not found
If this is not expected, reinstall Horovod with HOROVOD_WITH_PYTORCH=1 to debug the build error.
Warning! MPI libs are missing, but python applications are still available.
```

Reply:
> @romerojosh I was using the same set of commands to build the image with OpenMPI 4.1.4 on my mac.

Our CI tests Horovod on mac with OpenMPI 4.1.1_1-1. All it does is: `brew install -f openmpi`:
https://github.com/horovod/horovod/runs/8077816985?check_suite_focus=true#step:3:111

@kvignesh1420 do install Horovod with `HOROVOD_WITH_MPI=1`? This tells Horovod to check that it compiles against MPI of fail if it cannot find it: https://horovod.readthedocs.io/en/stable/install_include.html#mpi

Reply:
@EnricoMi I am using the `Dockerfile.test.cpu` to build an image with OpenMPI4.1.4 on my mac. Running the test within that container throws the Illegal instruction error as per : https://github.com/horovod/horovod/pull/3662#issuecomment-1228885412

Didn't face this issue with OpenMPI4.0.0 though!